mlir/lib/Conversion/MIOpenToGPU/MIOpenToGPU.cpp

```diff
@@ -213,6 +228,34 @@ void LowerMIOpenOpsToGPUPass::runOnOperation() {
     b.setInsertionPointToEnd(&gpuFuncEntry);
     b.create<BranchOp>(loc, clonedFuncEntry);
 
+    // convert all calls to gpu.launch_func
+    SmallVector<CallOp, 4> calls;
+    op.walk([&](CallOp call) {
+      if (auto callable = call.getCallableForCallee()) {
+        if (FlatSymbolRefAttr symRef = callable.dyn_cast<SymbolRefAttr>()
+                                           .dyn_cast<FlatSymbolRefAttr>()) {
+          if (symRef.getValue() == theFunc.getName()) {
+            OpBuilder b(call);
```

Comment:
Prefer not to affect external builders. This is lightweight and sets insertion point.

Reply:
https://mlir.llvm.org/doxygen/classmlir_1_1OpBuilder_1_1InsertionGuard.html is presumably an object for a reason

Reply:
Lambda is meant to be a Functional Programming mechanism. Prefer to be additive only.

Reply:
Fair enough. At the very least we should pull the listener through since it presumably does _something_.

And I don't think these lambdas are meant to be stateless, necessarily.

Reply:
Actually will apply to f32 as well but just means smaller numbers.